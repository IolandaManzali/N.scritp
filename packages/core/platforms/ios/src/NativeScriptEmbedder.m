#import "NativeScriptEmbedder.h"

@implementation NativeScriptEmbedder

+ (NativeScriptEmbedder *)sharedInstance {
    static NativeScriptEmbedder *sharedInstance = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        sharedInstance = [[NativeScriptEmbedder alloc] init];
    });
    
    return sharedInstance;
}

- (void)setDelegate:(id <NativeScriptEmbedderDelegate>)aDelegate {
    _delegate = aDelegate;
}

- (void)setWindowScene:(UIWindowScene *)windowScene {
    NSLog(@"Found Window Scene!");
    _windowScene = windowScene;
}

+(void)setup {
    Class klass = NSClassFromString(@"NativeScriptStart");
    if (klass) {
        [klass setup];
    }
}

+(void)boot {
    Class klass = NSClassFromString(@"NativeScriptStart");
    if (klass) {
        [klass boot];
    }
}

@end
