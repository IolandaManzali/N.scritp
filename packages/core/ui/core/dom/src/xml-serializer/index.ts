/* eslint-disable no-case-declarations */
import Element from '../nodes/element/Element';
import VoidElements from '../config/VoidElements';
import DocumentType from '../nodes/document-type/DocumentType';
import { escape } from 'html-escaper';
import Node from '../nodes/node/Node';
import HTMLTemplateElement from '../nodes/html-template-element/HTMLTemplateElement';

/**
 * Utility for converting an element to string.
 * https://github.com/capricorn86/happy-dom/blob/master/packages/happy-dom/src/xml-serializer/XMLSerializer.ts
 */
export default class XMLSerializer {
	/**
	 * Renders an element as HTML.
	 *
	 * @param root Root element.
	 * @param [options] Options.
	 * @param [options.includeShadowRoots] Set to "true" to include shadow roots.
	 * @returns Result.
	 */
	public serializeToString(root: Node, options?: { includeShadowRoots?: boolean; innerHTML?: boolean }): string {
		const rootNode = (root as Element).tagName === 'TEMPLATE' ? (<HTMLTemplateElement>root).content : root;
		switch (root.nodeType) {
			case Node.ELEMENT_NODE:
				const element = <Element>rootNode;
				const tagName = element.tagName.toLowerCase();

				if (VoidElements.includes(tagName)) {
					return `<${tagName}${this._getAttributes(element)}>`;
				}

				let innerHTML = '';
				let currentNode = element.firstChild;
				while (currentNode) {
					innerHTML += this.serializeToString(currentNode, { ...options, innerHTML: false });
					currentNode = currentNode.nextSibling;
				}

				if (options?.includeShadowRoots && element.shadowRoot) {
					innerHTML += `<template shadowroot="${element.shadowRoot.mode}">`;

					for (const node of element.shadowRoot.childNodes) {
						innerHTML += this.serializeToString(node, { ...options, innerHTML: false });
					}

					innerHTML += '</template>';
				}
				if (options.innerHTML) {
					return innerHTML;
				} else {
					return `<${tagName}${this._getAttributes(element)}>${innerHTML}</${tagName}>`;
				}
			case Node.DOCUMENT_FRAGMENT_NODE:
			case Node.DOCUMENT_NODE: {
				let html = '';
				let currentNode = rootNode.firstChild;
				while (currentNode) {
					html += this.serializeToString(currentNode, { ...options, innerHTML: false });
					currentNode = currentNode.nextSibling;
				}
				return html;
			}
			case Node.COMMENT_NODE:
				return `<!--${root._textContent}-->`;
			case Node.TEXT_NODE:
				return root['textContent'];
			case Node.DOCUMENT_TYPE_NODE:
				const doctype = <DocumentType>root;
				const identifier = doctype.publicId ? ' PUBLIC' : doctype.systemId ? ' SYSTEM' : '';
				const publicId = doctype.publicId ? ` "${doctype.publicId}"` : '';
				const systemId = doctype.systemId ? ` "${doctype.systemId}"` : '';
				return `<!DOCTYPE ${doctype.name}${identifier}${publicId}${systemId}>`;
		}

		return '';
	}

	/**
	 * Returns attributes as a string.
	 *
	 * @param element Element.
	 * @returns Attributes.
	 */
	private _getAttributes(element: Element): string {
		let attributeString = '';

		// if (!(<Element>element)._attributes.is && (<Element>element)._isValue) {
		// 	attributeString += ' is="' + escape((<Element>element)._isValue) + '"';
		// }

		for (const attribute of (<Element>element).attributes) {
			if (attribute.value !== null) {
				attributeString += ' ' + attribute.name + '="' + escape(attribute.value) + '"';
			}
		}
		return attributeString;
	}
}
