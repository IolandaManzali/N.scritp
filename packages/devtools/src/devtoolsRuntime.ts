// const domainDispatchers = new Map<string, any>();

export function DomainDispatcher(domain: string): ClassDecorator {
	return (klass) => {
		__registerDomainDispatcher(domain, klass);
		// if (!domainDispatchers.has(domain)) {
		// 	domainDispatchers.set(domain, klass);
		// } else {
		// 	console.trace(`Domain dispatcher for ${domain} already registered!`);
		// }
	};
}

interface ProtocolMessage<T = any> {
	id: number;
	method: string;
	params: T;
}

export class ProtocolWrapper {
	constructor() {
		return ProtocolWrapper.wrap(this);
	}

	protected enabled: boolean = false;

	enable() {
		this.enabled = true;
	}

	disable() {
		this.enabled = false;
	}

	// private _enabled: boolean = false;

	// get enabled() {
	// 	return this._enabled;
	// }

	// enable() {
	// 	console.log(this.constructor.name, 'enable!');
	// 	this._enabled = true;
	// }

	// disable() {
	// 	this._enabled = false;
	// }

	public emit(name: string, params: any) {
		try {
			// console.info('[emit]', { method: name, params });
			__inspectorSendEvent(
				JSON.stringify({
					method: name,
					params,
				})
			);
		} catch (err) {
			console.error(err);
		}
	}

	public timestamp(): number {
		return __inspectorTimestamp();
	}

	private static async sendResponseToDevtools<T = any>(id: number, data: T) {
		try {
			const result = await data;
			if (result) {
				// console.info('[sendResponseToDevtools]', { id, result });
				__inspectorSendEvent(
					JSON.stringify({
						id,
						result,
					})
				);
			}
		} catch (err) {
			console.error(err);
		}
	}

	private static wrap<T extends Object>(handler: T): T {
		return new Proxy(handler, {
			get(target, prop) {
				if (typeof target[prop] === 'function') {
					return (params, message: ProtocolMessage) => {
						try {
							// console.warn('[incoming]', {
							// 	id: message.id,
							// 	method: message.method,
							// 	params,
							// 	message,
							// });
							const res = target[prop](params, message);
							ProtocolWrapper.sendResponseToDevtools(message.id, res);
							return res;
						} catch (err) {
							console.error(err);
						}
					};
				}
			},
		}) as T;
	}
}
