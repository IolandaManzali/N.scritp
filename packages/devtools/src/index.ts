import { ApplicationSettingsKeyValueProvider } from './impl/ApplicationSettingsKeyValueProvider';
import { KeyValueProviderRegistry } from './providers/keyValueProvider';

/**
 * Initializes the default devtools providers.
 */
export function initDevtools() {
	KeyValueProviderRegistry.registerKeyValueProvider(new ApplicationSettingsKeyValueProvider());
}

export { KeyValueProviderRegistry, KeyValueProvider } from './providers/keyValueProvider';
// export { SqlDatabaseProviderRegistry, SqlDatabaseProvider, SqlDatabaseResult } from './providers/sqlDatabaseProvider';
// export { IndexedDBDatabaseProviderRegistry, IndexedDBDatabaseProvider, IndexedDBDatabaseResult } from './providers/indexedDbDatabaseProvider';
// export { NetworkProviderRegistry, NetworkProvider } from './providers/networkProvider';

export { DomainDispatcher, ProtocolWrapper } from './devtoolsRuntime';
