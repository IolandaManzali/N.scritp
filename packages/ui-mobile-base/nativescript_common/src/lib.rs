#![feature(unix_chown)]
#![feature(io_error_more)]
extern crate core;
#[cfg(any(target_os = "ios", target_os = "macos"))]
#[macro_use]
extern crate objc;

pub mod common;

#[cfg(target_os = "android")]
#[allow(non_snake_case)]
pub mod android;

#[cfg(any(target_os = "ios", target_os = "macos"))]
pub mod ios;
