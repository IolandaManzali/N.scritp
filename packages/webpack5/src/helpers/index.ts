import { merge } from 'webpack-merge';

import { addCopyRule, removeCopyRule } from './copyRules';
import { determineProjectFlavor, projectUsesCustomFlavor } from './flavor';
import { error, info, warn } from './log';
import { getValue } from './config';
import {
	getAllDependencies,
	hasDependency,
	getDependencyPath,
} from './dependencies';
import { getPackageJson, getProjectRootPath } from './project';
import {
	addPlatform,
	getAbsoluteDistPath,
	getDistPath,
	getEntryDirPath,
	getEntryPath,
	getPlatform,
	getPlatformName,
} from './platform';
import { readTsConfig } from './typescript';

// intentionally populated manually
// as this generates nicer typings
// that show all the utils inline
// rather than imports to types
// todo: maybe use api-extractor instead
export default {
	merge,
	addCopyRule,
	removeCopyRule,
	config: {
		getValue,
	},
	dependencies: {
		getAllDependencies,
		hasDependency,
		getDependencyPath,
	},
	flavor: {
		determineProjectFlavor,
		projectUsesCustomFlavor
	},
	log: {
		error,
		info,
		warn,
	},
	project: {
		getProjectRootPath,
		getPackageJson,
	},
	platform: {
		addPlatform,
		getAbsoluteDistPath,
		getDistPath,
		getEntryDirPath,
		getEntryPath,
		getPlatform,
		getPlatformName,
	},
	tsconfig: {
		readTsConfig,
	},
};
