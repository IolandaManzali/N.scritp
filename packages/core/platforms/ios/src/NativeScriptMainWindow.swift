import SwiftUI
import NativeScriptEmbedder
import UIKit

struct NativeScriptMainWindow: Scene {
    var body: some Scene {
        // The main window that presents the app's modules.
        WindowGroup {
            NativeScriptAppView(found: { windowScene in
                NativeScriptEmbedder.sharedInstance().setWindowScene(windowScene)
            }).onAppear {
                print("NativeScriptAppView onAppear")
                DispatchQueue.main.async {
                    NativeScriptStart.boot()
                }
            }
        }
        .windowStyle(.plain)
    }

    init() {
        NativeScriptEmbedder.sharedInstance().setDelegate(NativeScriptViewRegistry.shared)
        NativeScriptStart.setup()
    }
}

// NOTE: this likely should go into @nativescript/swift-ui:
struct NativeScriptAppView: UIViewRepresentable {
    /// A closure that's called when the window is found.
    var found: ((UIWindowScene?) -> Void)
  
    func makeUIView(context: Context) -> UIView {
        print("make NativeScriptAppView")
        var window: UIWindow? {
            guard let scene = UIApplication.shared.connectedScenes.first,
                  let windowSceneDelegate = scene.delegate as? UIWindowSceneDelegate,
                  let window = windowSceneDelegate.window else {
                return nil
            }
            return window
        }
        
        DispatchQueue.main.async {
            found(window?.windowScene)
        }
        
        return NativeScriptViewRegistry.app.view
    }
    
    func updateUIView(_ uiView: UIView, context: Context) {
        print("NativeScriptAppView updateUIView")
        // could update data through the controller
    }
}

@objc public class NativeScriptViewRegistry: NSObject, NativeScriptEmbedderDelegate {
    @objc public static let shared = NativeScriptViewRegistry()
    @objc public static let app = UIKitContainerCtrl()

    @objc public let views = NSMutableDictionary()
    @objc public var callback: ((String, UIKitContainerView) -> Void)? = nil
    
    @objc public static func getKeyWindow() -> UIWindow? {
        return UIApplication
            .shared
            .connectedScenes
            .compactMap { ($0 as? UIWindowScene)?.keyWindow }
            .last
    }
    
    public func presentNativeScriptApp(_ vc: UIViewController!) -> Any! {
        vc.view.frame = NativeScriptViewRegistry.app.view.bounds
        vc.view.autoresizingMask = [.flexibleWidth, .flexibleHeight]
        NativeScriptViewRegistry.app.addChild(vc)
        NativeScriptViewRegistry.app.view.addSubview(vc.view)
        vc.didMove(toParent: NativeScriptViewRegistry.app)
        return NativeScriptViewRegistry.app
    }
}

// UIViewController
@objc public class UIKitContainerCtrl: UIViewController {
    // allow NativeScript to override updateData for custom handling
    @objc public var updateData: ((_ data: NSMutableDictionary) -> Void)? = nil
}

// UIView
@objc public class UIKitContainerView: UIView {
    // allow NativeScript to override updateData for custom handling
    @objc public var updateData: ((_ data: NSMutableDictionary) -> Void)? = nil
}

// Notes:
// importing RealityKit causes App protocol to fail:
// Type 'VisionProApp' does not conform to protocol 'App'
// (likely expected, just interesting)
//import RealityKit
