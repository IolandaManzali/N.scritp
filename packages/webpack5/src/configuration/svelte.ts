import base from './base';
import { env as _env, IWebpackEnv } from '@nativescript/webpack';
import Config from 'webpack-chain';
import { getPlatform } from '../helpers/project';
import { merge } from 'webpack-merge';
import svelteNativePreprocessor from 'svelte-native-preprocessor';

export default function (config: Config, env: IWebpackEnv = _env): Config {
	base(config, env);

	const platform = getPlatform();
	const mode = env.production ? 'production' : 'development';
	const production = mode === 'production';

	// resolve .svelte files
	// the order is reversed because we are using prepend!
	config.resolve.extensions.prepend('.svelte').prepend(`.${platform}.svelte`);

	// add a rule for .svelte files
	config.module
		.rule('svelte')
		.test(/\.svelte$/)
		.use('svelte-loader-hot')
		.loader('svelte-loader-hot')
		.tap((options) => {
			return {
				...options,
				// todo: should be a compiler object
				// but we want it as an external dependency
				dev: production,
				preprocess: [svelteNativePreprocessor()],
				hotReload: production,
				hotOptions: {
					injectCss: false,
					native: true,
				},
			};
		})
		.end();

	// set up ts support in svelte files
	config.module
		.rule('ts')
		.use('ts-loader')
		.loader('ts-loader')
		.tap((options = {}) => {
			return merge(options, {
				appendTsSuffixTo: ['\\.svelte$'],
			});
		});

	// add an alias for svelte, since some plugins may try to import it
	config.resolve.alias.set('svelte', 'svelte-native');

	return config;
}
