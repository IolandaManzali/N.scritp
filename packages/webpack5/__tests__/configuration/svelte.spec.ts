import Config from 'webpack-chain';

import {
	mockExistsSync,
	restoreExistsSync,
	setHasTSConfig,
} from '../../scripts/jest.utils';

import svelte from '../../src/configuration/svelte';
import { init } from '../../src';

jest.mock('__jest__/svelte.config.js', () => {}, { virtual: true });

describe('svelte configuration', () => {
	const platforms = ['ios', 'android'];

	beforeAll(() => {
		mockExistsSync();
		setHasTSConfig(true);
	});

	afterAll(() => {
		restoreExistsSync();
	});

	for (let platform of platforms) {
		it(`for ${platform}`, () => {
			init({
				[platform]: true,
			});
			expect(svelte(new Config()).toString()).toMatchSnapshot();
		});
	}
});
