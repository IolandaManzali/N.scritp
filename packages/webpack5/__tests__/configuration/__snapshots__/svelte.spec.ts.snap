// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`svelte configuration for android 1`] = `
"{
  mode: 'development',
  externals: [
    'package.json'
  ],
  devtool: 'inline-source-map',
  target: 'node',
  output: {
    path: '__jest__/platforms/android/app/src/main/assets/app',
    pathinfo: false,
    publicPath: '',
    libraryTarget: 'commonjs',
    globalObject: 'global'
  },
  resolve: {
    symlinks: true,
    alias: {
      '~/package.json': 'package.json',
      '~': '<TODO>appFullPath',
      '@': '<TODO>appFullPath',
      svelte: 'svelte-native'
    },
    extensions: [
      '.android.svelte',
      '.svelte',
      '.android.ts',
      '.ts',
      '.android.js',
      '.js',
      '.android.css',
      '.css',
      '.android.scss',
      '.scss',
      '.android.json',
      '.json'
    ]
  },
  resolveLoader: {
    modules: [
      'node_modules/@nativescript/webpack/dist/loaders',
      'node_modules'
    ]
  },
  module: {
    rules: [
      /* config.module.rule('ts') */
      {
        test: [
          /\\\\.ts$/
        ],
        use: [
          /* config.module.rule('ts').use('ts-loader') */
          {
            loader: 'ts-loader',
            options: {
              transpileOnly: true,
              allowTsInNodeModules: true,
              compilerOptions: {
                sourceMap: true,
                declaration: false
              },
              getCustomTransformers: function () { /* omitted long function */ },
              appendTsSuffixTo: [
                '\\\\\\\\.svelte$'
              ]
            }
          }
        ]
      },
      /* config.module.rule('js') */
      {
        test: /\\\\.js$/,
        use: [
          /* config.module.rule('js').use('babel-loader') */
          {
            loader: 'babel-loader'
          }
        ]
      },
      /* config.module.rule('css') */
      {
        test: /\\\\.css$/,
        use: [
          /* config.module.rule('css').use('apply-css-loader') */
          {
            loader: 'apply-css-loader'
          },
          /* config.module.rule('css').use('css2json-loader') */
          {
            loader: 'css2json-loader'
          }
        ]
      },
      /* config.module.rule('scss') */
      {
        test: /\\\\.scss$/,
        use: [
          /* config.module.rule('scss').use('apply-css-loader') */
          {
            loader: 'apply-css-loader'
          },
          /* config.module.rule('scss').use('css2json-loader') */
          {
            loader: 'css2json-loader'
          },
          /* config.module.rule('scss').use('scss-loader') */
          {
            loader: 'scss-loader'
          }
        ]
      },
      /* config.module.rule('svelte') */
      {
        test: /\\\\.svelte$/,
        use: [
          /* config.module.rule('svelte').use('svelte-loader-hot') */
          {
            loader: 'svelte-loader-hot',
            options: {
              dev: false,
              preprocess: [
                {
                  markup: function () { /* omitted long function */ }
                }
              ],
              hotReload: false,
              hotOptions: {
                injectCss: false,
                'native': true
              }
            }
          }
        ]
      }
    ]
  },
  optimization: {
    splitChunks: {
      cacheGroups: {
        defaultVendor: {
          test: /[\\\\\\\\/]node_modules[\\\\\\\\/]/,
          priority: -10,
          name: 'vendor',
          chunks: 'all'
        }
      }
    },
    minimizer: [
      /* config.optimization.minimizer('TerserPlugin') */
      new TerserPlugin(
        {
          terserOptions: {
            compress: {
              collapse_vars: false,
              sequences: false
            },
            keep_fnames: true
          }
        }
      )
    ]
  },
  plugins: [
    /* config.plugin('CleanWebpackPlugin') */
    new CleanWebpackPlugin(
      {
        cleanOnceBeforeBuildPatterns: [
          '__jest__/platforms/android/app/src/main/assets/app/**/*'
        ],
        verbose: false
      }
    ),
    /* config.plugin('DefinePlugin') */
    new DefinePlugin(
      {
        __DEV__: true,
        __NS_WEBPACK__: true,
        __CSS_PARSER__: '\\"css-tree\\"',
        __ANDROID__: true,
        __IOS__: false,
        'global.isAndroid': true,
        'global.isIOS': false,
        process: 'global.process',
        profile: '() => {}'
      }
    ),
    /* config.plugin('BundleAnalyzerPlugin') */
    new BundleAnalyzerPlugin(),
    /* config.plugin('WatchStateLoggerPlugin') */
    new WatchStateLoggerPlugin()
  ],
  entry: {
    bundle: [
      'src/app.js'
    ]
  }
}"
`;

exports[`svelte configuration for ios 1`] = `
"{
  mode: 'development',
  externals: [
    'package.json'
  ],
  devtool: 'inline-source-map',
  target: 'node',
  output: {
    path: '__jest__/platforms/ios/__jest__/app',
    pathinfo: false,
    publicPath: '',
    libraryTarget: 'commonjs',
    globalObject: 'global'
  },
  resolve: {
    symlinks: true,
    alias: {
      '~/package.json': 'package.json',
      '~': '<TODO>appFullPath',
      '@': '<TODO>appFullPath',
      svelte: 'svelte-native'
    },
    extensions: [
      '.ios.svelte',
      '.svelte',
      '.ios.ts',
      '.ts',
      '.ios.js',
      '.js',
      '.ios.css',
      '.css',
      '.ios.scss',
      '.scss',
      '.ios.json',
      '.json'
    ]
  },
  resolveLoader: {
    modules: [
      'node_modules/@nativescript/webpack/dist/loaders',
      'node_modules'
    ]
  },
  module: {
    rules: [
      /* config.module.rule('ts') */
      {
        test: [
          /\\\\.ts$/
        ],
        use: [
          /* config.module.rule('ts').use('ts-loader') */
          {
            loader: 'ts-loader',
            options: {
              transpileOnly: true,
              allowTsInNodeModules: true,
              compilerOptions: {
                sourceMap: true,
                declaration: false
              },
              getCustomTransformers: function () { /* omitted long function */ },
              appendTsSuffixTo: [
                '\\\\\\\\.svelte$'
              ]
            }
          }
        ]
      },
      /* config.module.rule('js') */
      {
        test: /\\\\.js$/,
        use: [
          /* config.module.rule('js').use('babel-loader') */
          {
            loader: 'babel-loader'
          }
        ]
      },
      /* config.module.rule('css') */
      {
        test: /\\\\.css$/,
        use: [
          /* config.module.rule('css').use('apply-css-loader') */
          {
            loader: 'apply-css-loader'
          },
          /* config.module.rule('css').use('css2json-loader') */
          {
            loader: 'css2json-loader'
          }
        ]
      },
      /* config.module.rule('scss') */
      {
        test: /\\\\.scss$/,
        use: [
          /* config.module.rule('scss').use('apply-css-loader') */
          {
            loader: 'apply-css-loader'
          },
          /* config.module.rule('scss').use('css2json-loader') */
          {
            loader: 'css2json-loader'
          },
          /* config.module.rule('scss').use('scss-loader') */
          {
            loader: 'scss-loader'
          }
        ]
      },
      /* config.module.rule('svelte') */
      {
        test: /\\\\.svelte$/,
        use: [
          /* config.module.rule('svelte').use('svelte-loader-hot') */
          {
            loader: 'svelte-loader-hot',
            options: {
              dev: false,
              preprocess: [
                {
                  markup: function () { /* omitted long function */ }
                }
              ],
              hotReload: false,
              hotOptions: {
                injectCss: false,
                'native': true
              }
            }
          }
        ]
      }
    ]
  },
  optimization: {
    splitChunks: {
      cacheGroups: {
        defaultVendor: {
          test: /[\\\\\\\\/]node_modules[\\\\\\\\/]/,
          priority: -10,
          name: 'vendor',
          chunks: 'all'
        }
      }
    },
    minimizer: [
      /* config.optimization.minimizer('TerserPlugin') */
      new TerserPlugin(
        {
          terserOptions: {
            compress: {
              collapse_vars: true,
              sequences: true
            },
            keep_fnames: true
          }
        }
      )
    ]
  },
  plugins: [
    /* config.plugin('CleanWebpackPlugin') */
    new CleanWebpackPlugin(
      {
        cleanOnceBeforeBuildPatterns: [
          '__jest__/platforms/ios/__jest__/app/**/*'
        ],
        verbose: false
      }
    ),
    /* config.plugin('DefinePlugin') */
    new DefinePlugin(
      {
        __DEV__: true,
        __NS_WEBPACK__: true,
        __CSS_PARSER__: '\\"css-tree\\"',
        __ANDROID__: false,
        __IOS__: true,
        'global.isAndroid': false,
        'global.isIOS': true,
        process: 'global.process',
        profile: '() => {}'
      }
    ),
    /* config.plugin('BundleAnalyzerPlugin') */
    new BundleAnalyzerPlugin(),
    /* config.plugin('WatchStateLoggerPlugin') */
    new WatchStateLoggerPlugin()
  ],
  entry: {
    bundle: [
      'src/app.js'
    ],
    'tns_modules/@nativescript/core/inspector_modules': [
      '@nativescript/core/inspector_modules'
    ]
  }
}"
`;
